docs/guides/using-requirements-file.rst

```diff
@@ -0,0 +1,117 @@
+ - You are using external packages, extensions, and themes that you need to include in your documentation project
+ - You have enabled the install project in a virtual environment option, but your package's setup.py file is not in the root directory.
+ - You have multiple python packages in your project and you wish to document them.
+
+-----------------------------
+Using a pip requirements file
+-----------------------------
+
+Creating the requirements file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You are using external packages, extensions, and themes that you need to include in your documentation project
+---------------------------------------------------------------------------------------------------------------
+
+Read the Docs supports adding specific functionality, and themes to tailor the behavior and appearance
+of your project's documentation. This is done by specifying a list of the packages, extensions, themes, to be used,
+and their versions, during the documentation build process.
+
+For example, one can specify which version of *Sphinx* to use, which theme they would like to use,
+what extensions they would like to add and so on.
+
+To do this, one can create a list of requirements and save it in a requirements file.
+
+Since *RTD* uses Python language to build the workflow which automatically converts and hosts your project's
+documentation, and uses various python packages for this, the requirments file is the same as any standard python project.
+Read the Docs installs them in a virtual environment using pip, the standard python package manager.
+
+In-fact, it is helpful to think of your documentation on readthedocs as a sub-project written in python,
+within your main project's repo.
+
+To use the requirements file:
+Create a text file with a sensible name in the root directory of your documentation directory. For example::
+
+    docs/requirements.txt
+    docs/requirements-docs.txt
+
+This is a standard python requirements file. If you know how to use that, this is the same thing. If not, read on.
+In this file, list all the packages (one package per line) that you require for building your documentation.
+Make sure to specify their appropriate version.
+
+For example, say you wish to only use Sphinx version 1.1.x and the sphinx_rtd_theme with a minimum version of 0.1.9.
+You are also using external extensions, for example, the napolean extension, (make sure to specify them in
+the extensions list in conf.py file), then your requirements file might look something like this:
+
+::
+
+	# <insert optional comment explaining why this package or version is a requirement>
+	sphinx == 1.1.*
+	sphinx_rtd_theme >= 0.1.9
+	sphinxcontrib-napolean
+
+
+You have enabled the install project in a virtual environment option, but your package's setup.py file is not in the root directory.
+---------------------------------------------------------------------------------------------------------------------------------------
+OR
```

Comment:
These are headers with permalinks. A shorter heading would be better.

![screen shot 2018-02-20 at 4 16 19 pm](https://user-images.githubusercontent.com/185043/36456392-85d2cbbe-1659-11e8-9ae9-822805ff04ed.png)